src/main/java/org/folio/service/orders/lines/update/OrderLinePatchOperationService.java

```diff
@@ -63,17 +64,21 @@ public class OrderLinePatchOperationService {
   private final PurchaseOrderLineService purchaseOrderLineService;
 
   private final InventoryCache inventoryCache;
+  private final InventoryManager inventoryManager;
 
   public OrderLinePatchOperationService(RestClient restClient, OrderLinePatchOperationHandlerResolver orderLinePatchOperationHandlerResolver,
-                                        PurchaseOrderLineService purchaseOrderLineService, InventoryCache inventoryCache) {
+                                        PurchaseOrderLineService purchaseOrderLineService, InventoryCache inventoryCache, InventoryManager inventoryManager) {
     this.restClient = restClient;
     this.orderLinePatchOperationHandlerResolver = orderLinePatchOperationHandlerResolver;
     this.purchaseOrderLineService = purchaseOrderLineService;
     this.inventoryCache = inventoryCache;
+    this.inventoryManager = inventoryManager;
   }
 
   public Future<Void> patch(String lineId, PatchOrderLineRequest request, RequestContext requestContext) {
-    return patchOrderLine(request, lineId, requestContext)
+    String newInstanceId = request.getReplaceInstanceRef().getNewInstanceId();
+    return inventoryManager.createShadowInstanceIfNeeded(newInstanceId, requestContext)
```

Comment:
check if ECS is enabled before performing shadow instance operations.

Reply:
Done